### DISCUSSION TITLE
el-input emit doesn't work as expected

### DISCUSSION THREAD

**Discussion by steffenstolze** [element-plus/element-plus#10567] [Q&A]
Hey there,

I'm trying to emit data from the input event of a el-input component to a parent component. It doesn't work. I face no issues with a normal HTML5 Input.

It works with the v-model approach, but not with emitting the input events. Why?
The el-input control always emits only the last typed character, not the whole string.

Here is the example: [Element Plus Playground Example](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgdmVyc2lvbiBhcyBFcFZlcnNpb24gfSBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgRWxlbWVudElucHV0TW9kZWwgZnJvbSAnLi9FbGVtZW50SW5wdXRNb2RlbC52dWUnXG5pbXBvcnQgRWxlbWVudElucHV0IGZyb20gJy4vRWxlbWVudElucHV0LnZ1ZSdcbmltcG9ydCBOYXRpdmVJbnB1dCBmcm9tICcuL05hdGl2ZUlucHV0LnZ1ZSdcbiAgXG5jb25zdCBzZWFyY2hUZXh0ID0gcmVmKCd0ZXN0Jyk7XG5jb25zdCBzZWFyY2hUZXh0TW9kZWwgPSByZWYoJ3Rlc3QnKTtcbmNvbnN0IHNlYXJjaFRleHROYXRpdmUgPSByZWYoJ3Rlc3QnKTtcbiAgXG5mdW5jdGlvbiBvblVwZGF0ZVRleHQoZXZlbnQpe1xuICBzZWFyY2hUZXh0TmF0aXZlLnZhbHVlID0gZXZlbnQ7XG59XG4gIFxuZnVuY3Rpb24gb25VcGRhdGVFbGVtZW50VGV4dChldmVudCl7XG4gIHNlYXJjaFRleHQudmFsdWUgPSBldmVudDtcbn1cbiAgXG5cbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIFxuICA8ZGl2PlxuICAgIEVMRU1FTlQgUExVUyBXSVRIIE1PREVMXG4gXHRcdDxFbGVtZW50SW5wdXRNb2RlbCB2LW1vZGVsOnNlYXJjaC10ZXh0PVwic2VhcmNoVGV4dE1vZGVsXCIgc3R5bGU9XCJ3aWR0aDogMTAwcHg7XCIvPlxuICBcdHt7c2VhcmNoVGV4dE1vZGVsfX1cbiAgPC9kaXY+XG4gIFxuICBcbiAgPGRpdj5cbiAgICBFTEVNRU5UIFBMVVNcbiBcdFx0PEVsZW1lbnRJbnB1dCA6c2VhcmNoLXRleHQ9XCJzZWFyY2hUZXh0XCIgQHVwZGF0ZS10ZXh0PVwib25VcGRhdGVFbGVtZW50VGV4dFwiIHN0eWxlPVwid2lkdGg6IDEwMHB4O1wiLz5cbiAgXHR7e3NlYXJjaFRleHR9fVxuICA8L2Rpdj5cbiAgXG4gIFxuICA8ZGl2PiAgXG4gICAgTkFUSVZFXG4gXHRcdDxOYXRpdmVJbnB1dCA6c2VhcmNoLXRleHQ9XCJzZWFyY2hUZXh0TmF0aXZlXCIgQHVwZGF0ZS10ZXh0PVwib25VcGRhdGVUZXh0XCIgc3R5bGU9XCJ3aWR0aDogMTAwcHg7XCIvPlxuIFx0XHR7e3NlYXJjaFRleHROYXRpdmV9fVxuICA8L2Rpdj5cbiAgXG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJOYXRpdmVJbnB1dC52dWUiOiI8c2NyaXB0IHNldHVwPlxuaW1wb3J0IHsgZGVmaW5lUHJvcHMsIGRlZmluZUVtaXRzIH0gZnJvbSAndnVlJztcblxuY29uc3QgcHJvcHMgPSBkZWZpbmVQcm9wcyh7XG4gIHNlYXJjaFRleHQ6IFN0cmluZyxcbn0pO1xuY29uc3QgZW1pdCA9IGRlZmluZUVtaXRzKHtcbiAgdXBkYXRlVGV4dDogU3RyaW5nXG59KTtcblxuZnVuY3Rpb24gb25JbnB1dChldmVudCkge1xuICBlbWl0KCd1cGRhdGVUZXh0JywgZXZlbnQudGFyZ2V0LnZhbHVlKTtcbn1cbjwvc2NyaXB0PlxuPHRlbXBsYXRlPlxuICA8aW5wdXQgdHlwZT1cInRleHRcIiA6dmFsdWU9XCJzZWFyY2hUZXh0XCIgQGlucHV0PVwib25JbnB1dFwiIC8+ICAgXG48L3RlbXBsYXRlPlxuXG4iLCJFbGVtZW50SW5wdXQudnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCB7IGRlZmluZVByb3BzLCBkZWZpbmVFbWl0cyB9IGZyb20gJ3Z1ZSc7XG5pbXBvcnQgeyBFbElucHV0IH0gZnJvbSAnZWxlbWVudC1wbHVzJztcblxuY29uc3QgcHJvcHMgPSBkZWZpbmVQcm9wcyh7XG4gIHNlYXJjaFRleHQ6IFN0cmluZyxcbn0pO1xuY29uc3QgZW1pdCA9IGRlZmluZUVtaXRzKHtcbiAgdXBkYXRlVGV4dDogU3RyaW5nXG59KTtcblxuZnVuY3Rpb24gb25JbnB1dChldmVudCkge1xuICBlbWl0KCd1cGRhdGVUZXh0JywgZXZlbnQpO1xufVxuPC9zY3JpcHQ+XG48dGVtcGxhdGU+XG4gXHQ8ZWwtaW5wdXQgdHlwZT1cInRleHRcIiA6dmFsdWU9XCJzZWFyY2hUZXh0XCIgQGlucHV0PVwib25JbnB1dFwiIC8+XG48L3RlbXBsYXRlPlxuXG4iLCJFbGVtZW50SW5wdXRNb2RlbC52dWUiOiI8c2NyaXB0IHNldHVwPlxuaW1wb3J0IHsgZGVmaW5lUHJvcHMsIGRlZmluZUVtaXRzIH0gZnJvbSAndnVlJztcbmltcG9ydCB7IEVsSW5wdXQgfSBmcm9tICdlbGVtZW50LXBsdXMnO1xuXG5jb25zdCBwcm9wcyA9IGRlZmluZVByb3BzKHtcbiAgc2VhcmNoVGV4dDogU3RyaW5nLFxufSk7XG5jb25zdCBlbWl0ID0gZGVmaW5lRW1pdHMoWyd1cGRhdGU6c2VhcmNoVGV4dCddKTtcblxuPC9zY3JpcHQ+XG48dGVtcGxhdGU+XG4gXHQ8ZWwtaW5wdXQgdHlwZT1cInRleHRcIiA6bW9kZWwtdmFsdWU9XCJwcm9wcy5zZWFyY2hUZXh0XCIgQHVwZGF0ZTptb2RlbFZhbHVlPVwiZW1pdCgndXBkYXRlOnNlYXJjaFRleHQnLCAkZXZlbnQpXCIgLz5cbjwvdGVtcGxhdGU+XG5cbiIsIl9vIjp7fX0=)

**Comment by chenxch** [comment]
[demo](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgdmVyc2lvbiBhcyBFcFZlcnNpb24gfSBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgRWxlbWVudElucHV0TW9kZWwgZnJvbSAnLi9FbGVtZW50SW5wdXRNb2RlbC52dWUnXG5pbXBvcnQgRWxlbWVudElucHV0IGZyb20gJy4vRWxlbWVudElucHV0LnZ1ZSdcbmltcG9ydCBOYXRpdmVJbnB1dCBmcm9tICcuL05hdGl2ZUlucHV0LnZ1ZSdcbiAgXG5jb25zdCBzZWFyY2hUZXh0ID0gcmVmKCd0ZXN0Jyk7XG5jb25zdCBzZWFyY2hUZXh0TW9kZWwgPSByZWYoJ3Rlc3QnKTtcbmNvbnN0IHNlYXJjaFRleHROYXRpdmUgPSByZWYoJ3Rlc3QnKTtcbiAgXG5mdW5jdGlvbiBvblVwZGF0ZVRleHQoZXZlbnQpe1xuICBzZWFyY2hUZXh0TmF0aXZlLnZhbHVlID0gZXZlbnQ7XG59XG4gIFxuZnVuY3Rpb24gb25VcGRhdGVFbGVtZW50VGV4dChldmVudCl7XG4gIHNlYXJjaFRleHQudmFsdWUgPSBldmVudDtcbn1cbiAgXG5cbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIFxuICA8ZGl2PlxuICAgIEVMRU1FTlQgUExVUyBXSVRIIE1PREVMXG4gXHRcdDxFbGVtZW50SW5wdXRNb2RlbCB2LW1vZGVsOnNlYXJjaC10ZXh0PVwic2VhcmNoVGV4dE1vZGVsXCIgc3R5bGU9XCJ3aWR0aDogMTAwcHg7XCIvPlxuICBcdHt7c2VhcmNoVGV4dE1vZGVsfX1cbiAgPC9kaXY+XG4gIFxuICBcbiAgPGRpdj5cbiAgICBFTEVNRU5UIFBMVVNcbiBcdFx0PEVsZW1lbnRJbnB1dCA6c2VhcmNoLXRleHQ9XCJzZWFyY2hUZXh0XCIgQHVwZGF0ZS10ZXh0PVwib25VcGRhdGVFbGVtZW50VGV4dFwiIHN0eWxlPVwid2lkdGg6IDEwMHB4O1wiLz5cbiAgXHR7e3NlYXJjaFRleHR9fVxuICA8L2Rpdj5cbiAgXG4gIFxuICA8ZGl2PiAgXG4gICAgTkFUSVZFXG4gXHRcdDxOYXRpdmVJbnB1dCA6c2VhcmNoLXRleHQ9XCJzZWFyY2hUZXh0TmF0aXZlXCIgQHVwZGF0ZS10ZXh0PVwib25VcGRhdGVUZXh0XCIgc3R5bGU9XCJ3aWR0aDogMTAwcHg7XCIvPlxuIFx0XHR7e3NlYXJjaFRleHROYXRpdmV9fVxuICA8L2Rpdj5cbiAgXG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJOYXRpdmVJbnB1dC52dWUiOiI8c2NyaXB0IHNldHVwPlxuaW1wb3J0IHsgZGVmaW5lUHJvcHMsIGRlZmluZUVtaXRzIH0gZnJvbSAndnVlJztcblxuY29uc3QgcHJvcHMgPSBkZWZpbmVQcm9wcyh7XG4gIHNlYXJjaFRleHQ6IFN0cmluZyxcbn0pO1xuY29uc3QgZW1pdCA9IGRlZmluZUVtaXRzKHtcbiAgdXBkYXRlVGV4dDogU3RyaW5nXG59KTtcblxuZnVuY3Rpb24gb25JbnB1dChldmVudCkge1xuICBlbWl0KCd1cGRhdGVUZXh0JywgZXZlbnQudGFyZ2V0LnZhbHVlKTtcbn1cbjwvc2NyaXB0PlxuPHRlbXBsYXRlPlxuICA8aW5wdXQgdHlwZT1cInRleHRcIiA6dmFsdWU9XCJzZWFyY2hUZXh0XCIgQGlucHV0PVwib25JbnB1dFwiIC8+ICAgXG48L3RlbXBsYXRlPlxuXG4iLCJFbGVtZW50SW5wdXQudnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCB7IGRlZmluZVByb3BzLCBkZWZpbmVFbWl0cyB9IGZyb20gJ3Z1ZSc7XG5pbXBvcnQgeyBFbElucHV0IH0gZnJvbSAnZWxlbWVudC1wbHVzJztcblxuY29uc3QgcHJvcHMgPSBkZWZpbmVQcm9wcyh7XG4gIHNlYXJjaFRleHQ6IFN0cmluZyxcbn0pO1xuY29uc3QgZW1pdCA9IGRlZmluZUVtaXRzKHtcbiAgdXBkYXRlVGV4dDogU3RyaW5nXG59KTtcblxuZnVuY3Rpb24gb25JbnB1dChldmVudCkge1xuICBlbWl0KCd1cGRhdGVUZXh0JywgZXZlbnQpO1xufVxuPC9zY3JpcHQ+XG48dGVtcGxhdGU+XG4gXHQ8ZWwtaW5wdXQgdHlwZT1cInRleHRcIiA6bW9kZWwtdmFsdWU9XCJzZWFyY2hUZXh0XCIgQGlucHV0PVwib25JbnB1dFwiIC8+XG48L3RlbXBsYXRlPlxuXG4iLCJFbGVtZW50SW5wdXRNb2RlbC52dWUiOiI8c2NyaXB0IHNldHVwPlxuaW1wb3J0IHsgZGVmaW5lUHJvcHMsIGRlZmluZUVtaXRzIH0gZnJvbSAndnVlJztcbmltcG9ydCB7IEVsSW5wdXQgfSBmcm9tICdlbGVtZW50LXBsdXMnO1xuXG5jb25zdCBwcm9wcyA9IGRlZmluZVByb3BzKHtcbiAgc2VhcmNoVGV4dDogU3RyaW5nLFxufSk7XG5jb25zdCBlbWl0ID0gZGVmaW5lRW1pdHMoWyd1cGRhdGU6c2VhcmNoVGV4dCddKTtcblxuPC9zY3JpcHQ+XG48dGVtcGxhdGU+XG4gXHQ8ZWwtaW5wdXQgdHlwZT1cInRleHRcIiA6bW9kZWwtdmFsdWU9XCJwcm9wcy5zZWFyY2hUZXh0XCIgQHVwZGF0ZTptb2RlbFZhbHVlPVwiZW1pdCgndXBkYXRlOnNlYXJjaFRleHQnLCAkZXZlbnQpXCIgLz5cbjwvdGVtcGxhdGU+XG5cbiIsIl9vIjp7fX0=)

Just change the `:value` of the `ElementInput.vue` file to `:model-value`.
